EasyPost/BetaFeatures/Parameters/Parameters.cs

```diff
@@ -0,0 +1,186 @@
+            return _parameterDictionary.ToStringNonNullableObjectDictionary();
+        }
+
+        /// <summary>
+        ///     Add a parameter to the dictionary.
+        /// </summary>
+        /// <param name="requestParameterAttribute"><see cref="RequestParameterAttribute" /> of the parameter to add.</param>
+        /// <param name="value">The value of parameter.</param>
+        private void Add(RequestParameterAttribute requestParameterAttribute, object? value)
+        {
+            // Primitive types (i.e. strings, booleans) can be added directly to the dictionary
+
+            // If the value is an object type, we know by this point it must inherit the corresponding base-IParameter interface
+            // If we've done our job correctly, the only classes that inherit base-IParameter interfaces are base-EasyPostObject and base-Parameters
+
+            // If a given value is a base-EasyPostObject object, serialize it as a dictionary
+            if (value is EasyPostObject easyPostObject) value = easyPostObject.AsDictionary();
+
+            // If the given value is another base-Parameters object, serialize it as a sub-dictionary for the parent dictionary
+            // This is because the JSON schema for a sub-object is different than the JSON schema for a top-level object
+            // e.g. the schema for an address in the address create API call is different than the schema for an address in the shipment create API call
+            if (value is Parameters parameters) value = parameters.ToSubDictionary(GetType());
+
+            // Finally, add the value to the dictionary
+            _parameterDictionary = UpdateDictionary(_parameterDictionary, requestParameterAttribute.JsonPath, value);
+        }
+
+        /// <summary>
+        ///     Update a dictionary with a new value.
+        /// </summary>
+        /// <param name="dictionary">Dictionary to update.</param>
+        /// <param name="keys">Path of new value to add.</param>
+        /// <param name="value">New value to add.</param>
+        /// <returns>Updated dictionary.</returns>
+        /// <exception cref="Exception">Could not add value to dictionary.</exception>
+        [SuppressMessage("Style", "IDE0045:Convert to conditional expression", Justification = "Harder to read")]
+        private static Dictionary<string, object?> UpdateDictionary(Dictionary<string, object?>? dictionary, string[] keys, object? value)
+        {
+            dictionary ??= new Dictionary<string, object?>();
+
+            switch (keys.Length)
+            {
+                // Don't need to go down
+                case 0:
+                    return dictionary;
+
+                // Last key left
+                case 1:
+                    string soloKey = keys[0];
+                    if (dictionary.ContainsKey(soloKey))
+                        // Key-value pair already exists in dictionary (likely because of override parameters)
```

Comment:
Yeah I definitely don't agree with this `Single-line comment should be preceded by blank line` github actions checker rule. Can we please turn these off? They are incredibly noisy and we haven't come back to fix them. They are doing more harm during code reviews than good since we are ignoring them all anyways.